web/ui/src/features/component/TableHead.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from 'react';
+
+import { SortOrder } from './types';
+
+interface Props {
+  headers: string[];
+  handleSorting?: (header: string, sortOrder: SortOrder) => void;
+}
+
+const TableHead = ({ headers, handleSorting }: Props) => {
+  const [sortField, setSortField] = useState('');
+  const [order, setOrder] = useState(SortOrder.ASC);
+
+  const handleSortingChange = (header: string) => {
+    const sortOrder = header === sortField && order === SortOrder.ASC ? SortOrder.DESC : SortOrder.ASC;
```

Comment:
I found the logic a little bit hard to understand. Maybe we could refactor the code a bit to make the two following scenarios more obvious:
- the user clicks on a new header
- the user clicks on the same header another time

I would suggest this, although it is a bit verbose it clearly handles both scenarios:
```
if (header === sortField) {
  // If the user clicked again on the same header we toggle the sort order 
  sortOrder = order === SortOrder.ASC ? SortOrder.DESC : SortOrder.ASC;
} else {
  // If the user clicks on a new header we sort the entries by default in ASC order
  sortOrder = SortOrder.ASC;
}
```

Reply:
Done this one too in latest commit.